### ISSUE TITLE
wslg support?

### ISSUE THREAD

**Issue by zoeleu** [am11/VoidMuslWSL#6]

**Comment by am11** [comment]
Hi, can you say a few words what is the issue? Have you tried running GUI apps in VoidMuslWSL with wslg?

I haven't gotten a chance to explore wslg, it is only available in the insider build. I will try once it is shipped with GA bits.

However, reading this: https://github.com/microsoft/wslg#wslg-system-distro, I don't know if there is anything we need to change to make it work, but I'd defer to your findings.

**Comment by lumikalt** [comment]
With the bit of testing done by me, neither GTK nor QT appear to be working...
Tracebacks:

`QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
Could not initialize GLX
fish: Job 1, 'sudo cool-retro-term' terminated by signal SIGABRT (Abort)`

`(geany:2257): Gdk-WARNING **: 21:58:31.436: Native Windows taller than 65535 pixels are not supported
(geany:2257): Gdk-CRITICAL **: 21:58:31.470: ../gdk/wayland/gdkdisplay-wayland.c:1399: Unable to create Cairo image surface: invalid value (typically too big) for the size of the input (surface, pattern, etc.)
fish: Job 1, 'geany' terminated by signal SIGSEGV (Address boundary error)`

**Comment by lumikalt** [comment]
Of course, attempting to create a user and using sudo to elevate privileges also appears not to work - an issue I ran into due to dbus (or systemd?) not starting or something like that. Obviously there's no systemd here so that option is invalidated. Alas, everything else works perfectly.

**Comment by am11** [comment]
See https://github.com/am11/VoidWSL/issues/7. Does `export DISPLAY=:0` help?

**Comment by lumikalt** [comment]
Nope, for neither of the graphics libraries...

**Comment by am11** [comment]
I haven't tried WSLg myself. If it is broken on AlpineWSL - https://git.io/alpwsl -  as well, then it could be that some component is musl-libc incompatible? We don't do anything different in VoidWSL build script which is missed out in musl edition (or vice-versa).

@StellarOrbit, on a fresh VoidWSL installation, do you remember the minimal steps required to get basic GUI apps working?

Would be nice to have this information in README. PRs are welcome. :slightly_smiling_face:

**Comment by lumikalt** [comment]
Theoretically, Void should be able to use its powers to fix the incompatibilities, so this should indeed be a conflict between WSL and musl... Wonder how one would fix that.

**Comment by StellarOrbit** [comment]
> I haven't tried WSLg myself. If it is broken on AlpineWSL - https://git.io/alpwsl - as well, then it could be that some component is musl-libc incompatible? We don't do anything different in VoidWSL build script which is missed out in musl edition (or vice-versa).
> 
> @StellarOrbit, on a fresh VoidWSL installation, do you remember the minimal steps required to get basic GUI apps working?
> 
> Would be nice to have this information in README. PRs are welcome. 🙂

Apologies for the late reply. I know that Gedit doesn't work by default because it is missing something when just installing gedit and it fails to launch but other gui programs launch. I would have to go back and reinstall voidwsl to see what is the errors and how I overcame them.
